api/file-translations/params/base-parameters-with-callback.ts

```diff
@@ -0,0 +1,32 @@
+import { BaseParameters } from "../../parameters";
+import { FtsCallback, FtsCallbackMethod, FtsCallbackUserData } from "./fts-callback-parameter";
+
+export class BaseParametersWithCallback extends BaseParameters {
+    setCallback(
+        url: string,
+        httpMethod?: FtsCallbackMethod,
+        userData?: FtsCallbackUserData | string
+    ): this;
+    setCallback(callback: FtsCallback): this;
+
+    setCallback(
+        urlOrCallback: FtsCallback | string,
+        httpMethod: FtsCallbackMethod = FtsCallbackMethod.POST,
+        userData?: FtsCallbackUserData | string
```

Comment:
`urlOrCallback` - it's a php style( 3 options here:
1. `setCallback(callback: FtsCallback)`
2. `setCallback(url: string, httpMethod: FtsCallbackMethod = FtsCallbackMethod.POST, userData: FtsCallbackUserData)` (and define `FtsCallbackUserData` as `type FtsCallbackUserData = Record<string, unknown> | string;`, why do you need `... | string` in method signature?)
3. Introduce two methods with different parameters, but please don't merge them into one with union types.

Reply:
You also need to test all that cases. I doubt in real life you really need to support all that cases (pass as separate params or pass as an object etc). Let's simplify this as described above?

Reply:
1. removed all overloaded methods. left only one `setCallback(callback: FtsCallback)`
2. re-tested all cases on real requests